src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -231,5 +240,126 @@ static string GetContainerInlineContent(ContainerInline containerInline)
+                    }
+                    break;
+                default:
+                    break;
+            }
+            if (tableNode != null)
+            {
+                tableNodes.Add(tableNode);
+            }
+        }
+
+        private static TableNode TryParseTable(Table table)
+        {
+            if (table.Count == 0)
+            {
+                return new TableNode { IsSuccessParsed = false, };
+            }
+            var columnHeaders = new List<TableCellNode>();
+            var rowHeaders = new List<TableCellNode>();
+            var columnHeaderRow = (TableRow)table.First();
+            foreach (TableCell cell in columnHeaderRow)
+            {
+                columnHeaders.Add(ParseCell(cell));
+            }
+            foreach (TableRow row in table)
+            {
+                if (row.Count == 0)
+                {
+                    rowHeaders.Add(new TableCellNode { IsVisible = false });
+                }
+                else
+                {
+                    rowHeaders.Add(ParseCell((TableCell)row.First()));
+                }
+            }
+            return new TableNode
+            {
+                ColumnHeaders = columnHeaders.ToArray(),
+                RowHeaders = rowHeaders.ToArray(),
+                IsSuccessParsed = true,
+            };
+        }
+
+        private static TableCellNode ParseCell(TableCell cell)
+        {
+            if (cell == null || cell.Count == 0)
+            {
+                return new TableCellNode
+                {
+                    IsVisible = false,
+                    IsEmphasis = false,
+                };
+            }
+            var block = cell.LastChild;
+            var isVisible = MarkdigUtility.IsVisible(block);
+            var isEmphasis = false;
+            MarkdigUtility.Visit(block, node =>
+            {
+                var innerEmphasis = node switch
+                {
+                    ParagraphBlock paragraph when paragraph.Inline.FindDescendants<EmphasisInline>().Any() => true,
+                    _ => false,
+                };
+                return isEmphasis = isEmphasis || innerEmphasis;
+            });
+            var cellNode = new TableCellNode
+            {
+                IsVisible = isVisible,
+                IsEmphasis = isEmphasis,
+            };
+            return cellNode;
+        }
+
+        private static bool TryDetectTable(ParagraphBlock paragraph, MarkdownEngine markdownEngine)
+        {
+            if (!paragraph.Inline.FindDescendants<LineBreakInline>().Any())
+            {
+                return false;
+            }
+
+            var plainText = markdownEngine.ToPlainText(paragraph);
+
+            var lines = plainText.Split("\n").ToList();
+            bool tableHeaderExist = false;
+            int tableDelimiterExistLine = 0;
+            foreach (var line in lines)
+            {
+                if (line.Contains("|"))
```

Comment:
"-" line should follow up with the line contains "|", and we can break earlier.

var tableDelimiterExistsLine = 0;
foreach(var line in lines)
{
var cells = line.Split("|");
if (!cells.Any())
{
    tableDelimiterExistsLine = 0;
    continue;
}

if (tableDelimiterExistsLine ++ > 0 && cells.Any(c => c.contains ("-")))
    return true;
}